### PR TITLE
RadioButton/Checkbox: Add describedby as a prop

### PR DESCRIPTION
**Pull request by somethiiing**
for https://axeauditor.dequecloud.com/test-run/de95ac14-f88c-11ee-a30d-2b2389e5f0d8/issue/07ac1b62-fc82-11ee-a631-135645ea1d04?sortField=ordinal&sortDir=asc&page=0&issuesCount=&searchKeyword=1724726&searchOn=id&row=0

the fix that deque recommends are:
image

I already recommended to @jennyhe1120 that she already do the second recommendation (add the fieldname to the error message), but since deque _strongly_ recommends the first option, i created this POC.

I created an ariaProps object that can be reused between the different inputs (checkbox/radio/textinput/etc), i only put `aria-describedby` for now, because that's the only one we need. other suggested ones are: `aria-label`, `aria-labelledby`, `aria-hidden` but concluded we didnt need them yet.

the other option was the just manually map the prop `aria-describedby` onto the the component, but that felt unwieldy if we ever added more (like in some places we already use `aria-label`)

### CODE REVIEW

**Comment by somethiiing** [comment]
@christianvuerings can you give input on how you'd handle the issue described in the pr description? if this POC looks okay, i can add tests/update the remaining form input components

File: packages/syntax-core/src/Checkbox/Checkbox.stories.tsx [hunk 1]
```diff
@@ -49,7 +49,14 @@ const CheckboxInteractive = ({
   const handleChange = () => {
     setIsChecked(!isChecked);
   };
-  return <Checkbox checked={isChecked} onChange={handleChange} label={label} />;
+  return (
+    <Checkbox
+      ariaValues={{ "aria-describedby": "checkboxes" }}
```
Comment (christianvuerings) [inline packages/syntax-core/src/Checkbox/Checkbox.stories.tsx:54]: @somethiiing does the `checkboxes` id actually exist on this page? If not, could we add it so we can test the behavior?

File: packages/syntax-core/src/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -68,6 +70,12 @@ const Checkbox = ({
    * @defaultValue false
    */
   error?: boolean;
+  /**
+   * Whether or not there is an error with the input
```
Comment (christianvuerings) [inline packages/syntax-core/src/Checkbox/Checkbox.tsx:74]: @somethiiing the description doesn't make a lot of sense - can we make it more specific?

File: packages/syntax-core/src/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -68,6 +70,12 @@ const Checkbox = ({
...
+   *
+   * @defaultValue false
+   */
+  ariaValues?: ariaProps;
```
Comment (christianvuerings) [inline packages/syntax-core/src/Checkbox/Checkbox.tsx:78]: @somethiiing I would avoid passing in an object of props when possible. Just use `aria-describedby` instead.

Comment (christianvuerings) [reply packages/syntax-core/src/Checkbox/Checkbox.tsx:78]: Example as to why we want to avoid it:
image

**Review by christianvuerings — APPROVED** [review]
@somethiiing going to approve to unblock but please make sure to address comments before merging.

- [x] Add `checkboxes` (or other element which uses aria describedby in the story book)
- [x] Update docs
- [x] Flatten props
- [x] Add `aria-describedby` to `RadioButton` as well

File: packages/syntax-core/src/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (somethiiing) [reply packages/syntax-core/src/Checkbox/Checkbox.tsx:74]: oops, i had copied and pasted the above error to get the same formatting, but forgot to change the actual error message

**Pull request merged**